File: app/src/main/java/fr/free/nrw/commons/wikidata/mwapi/MwServiceError.java
package fr.free.nrw.commons.wikidata.mwapi;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import org.apache.commons.lang3.StringUtils;
import fr.free.nrw.commons.wikidata.model.BaseModel;

/**
 * Gson POJO for a MediaWiki API error.
 */
public class MwServiceError extends BaseModel {

     @Nullable private String code;
     @Nullable private String text;

    @NonNull public String getTitle() {
        return StringUtils.defaultString(code);
    }

    @NonNull public String getDetails() {
        return StringUtils.defaultString(text);
    }

    @Nullable
    public String getCode() {
        return code;
    }
}

File: app/src/main/java/fr/free/nrw/commons/wikidata/mwapi/MwServiceError.kt
package fr.free.nrw.commons.wikidata.mwapi

import fr.free.nrw.commons.wikidata.model.BaseModel
import org.apache.commons.lang3.StringUtils

/**
 * Gson POJO for a MediaWiki API error.
 */
class MwServiceError : BaseModel() {
    val code: String? = null
    private val text: String? = null

    val title: String
        get() = code ?: ""

    val details: String
        get() = text ?: ""
}
